src/main/kotlin/com/github/julianhille/sqldelight/jscipherdriver/JsSqlCipherDriver.kt

```diff
@@ -0,0 +1,207 @@
+    createOrGetStatement(identifier, sql).run {
+      bind(binders)
+      run()
+    }
+  }
+
+  private fun Statement.bind(binders: (SqlPreparedStatement.() -> Unit)?) = binders?.let {
+    val bound = SqlJsCipherStatement()
+    binders(bound)
+    println(bound.parameters)
+    bind(bound.parameters.toTypedArray())
+  }
+
+  override fun executeQuery(
+    identifier: Int?,
+    sql: String,
+    parameters: Int,
+    binders: (SqlPreparedStatement.() -> Unit)?
+  ): SqlCursor {
+    return createOrGetStatement(identifier, sql).run {
+      bind(binders)
+      SqlJsCipherCursor(iterate())
+    }
+  }
+
+  override fun newTransaction(): Transacter.Transaction {
+    val enclosing = transaction
+    val transaction = Transaction(enclosing)
+    this.transaction = transaction
+    if (enclosing == null) {
+      db.exec("BEGIN TRANSACTION")
+    }
+    return transaction
+  }
+
+  override fun currentTransaction() = transaction
+
+  private inner class Transaction(
+    override val enclosingTransaction: Transacter.Transaction?
+  ) : Transacter.Transaction() {
+    override fun endTransaction(successful: Boolean) {
+      if (enclosingTransaction == null) {
+        if (successful) {
+          db.exec("END TRANSACTION")
+        } else {
+          db.exec("ROLLBACK TRANSACTION")
+        }
+      }
+      transaction = enclosingTransaction
+    }
+  }
+
+  private fun createOrGetStatement(identifier: Int?, sql: String): Statement = db.prepare(sql)
+
+  fun getVersion(): Int = (db.pragma("user_version") as? Double)?.toInt() ?: 0
+
+  private fun setVersion(version: Int): Unit {
+    db.pragma("user_version=${version}")
+  }
+
+  fun migrateIfNeeded(
+    create: (SqlDriver) -> Unit,
+    upgrade: (SqlDriver, Int, Int) -> Unit,
+    version: Int
+  ) {
+    val initialVersion = getVersion()
+    println("soimething mumble mumble $initialVersion")
+    println("soimething mumble mumble $db")
+    println("soimething mumble mumble $transaction")
+    val driver = this
+    if (initialVersion == 0) {
+      transaction.run {
+        create(driver)
+        setVersion(version)
+      }
+    } else if (initialVersion != version) {
+      if (initialVersion > version) {
+          throw IllegalStateException("Database version $initialVersion newer than config version $version")
+        }
+        transaction.run {
+          upgrade(driver, initialVersion, version)
+          setVersion(version)
+        }
+    }
+  }
+}
+
+class SqlJsCipherStatement: SqlPreparedStatement {
+  val parameters = mutableListOf<Any?>()
+  override fun bindBytes(index: Int, bytes: ByteArray?) {
+    parameters.add(bytes)
+  }
+
+  override fun bindLong(index: Int, long: Long?) {
+    // We convert Long to Double because Kotlin's Double is mapped to JS number
+    // whereas Kotlin's Long is implemented as a JS object
+    parameters.add(long?.toDouble())
```

Comment:
No use of index?

Why no need?

https://github.com/cashapp/sqldelight/blob/cb699fcc19632a70deeb2930470bcf09db625a42/drivers/native-driver/src/nativeMain/kotlin/app/cash/sqldelight/driver/native/SqliterStatement.kt